flink-table-store-core/src/main/java/org/apache/flink/table/store/file/mergetree/compact/aggregate/FieldMinAgg.java

```diff
@@ -0,0 +1,123 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.table.store.file.mergetree.compact.aggregate;
+
+import org.apache.flink.table.data.DecimalData;
+import org.apache.flink.table.data.DecimalDataUtils;
+import org.apache.flink.table.data.GenericRowData;
+import org.apache.flink.table.data.RowData;
+import org.apache.flink.table.data.TimestampData;
+import org.apache.flink.table.types.logical.LogicalType;
+
+/** min aggregate a field of a row. */
+public class FieldMinAgg extends FieldAggregator {
+    public FieldMinAgg(LogicalType logicalType) {
+        super(logicalType);
+    }
+
+    @Override
+    void agg(
+            GenericRowData mergeRow,
+            RowData inRow,
+            RowData.FieldGetter[] fieldGetters,
+            int fieldPos) {
+        Object min;
+        Object mergeField = fieldGetters[fieldPos].getFieldOrNull(mergeRow);
+        Object inField = fieldGetters[fieldPos].getFieldOrNull(inRow);
+
+        if (mergeField == null || inField == null) {
+            min = (mergeField == null ? inField : mergeField);
+        } else {
+            // ordered by type root definition
+            switch (fieldType.getTypeRoot()) {
```

Comment:
Can we just extract a method `compare` to reuse logical for min and max? Maybe just in `RowDataUtils`.